finat/tensor_product.py

```diff
@@ -21,7 +21,14 @@ class TensorProductElement(FiniteElementBase):
     def __init__(self, factors):
         super(TensorProductElement, self).__init__()
         self.factors = tuple(factors)
-        assert all(fe.value_shape == () for fe in self.factors)
+
+        shapes = [fe.value_shape for fe in self.factors if fe.value_shape != ()]
+        if len(shapes) == 0:
+            self._value_shape = ()
+        elif len(shapes) == 1:
+            self._value_shape = shapes[0]
+        else:
+            raise NotImplementedError("Only one nonscalar factor permitted!")
```

Comment:
OK, so we're formalising that we're not going to tensor product (say) a 2-simplex with a 2-simplex and support the 4-d de Rham complex right now.